HHS/Head-Start-TTADP
# Fixes for grantee record page, export behavior, increase prod memory, Distinct Program Specialist List, Save Animation, date picker fixes

## Description of change

**TTAHUB-446:** Added distinct list of Program Specialists to Grantee search result. Fixed Grantee Type and added distinct Program Specialist list to Grantee Profile page.

**TTAHUB-410:** Saving an AR now shows a "fade-in" animation on the alert. The Review and Submit page now also shows a save alert at the bottom.

**TTAHUB-471:** Memory limit on production increased. Export limit changed to 12,000 reports.

**TTAHUB-436:** Message on CSVs now references March 17 instead of March 1

**TTAHUB-462:** Grant list table on Grantee Profile page should have rows with background colors. Target population table will show "0" entries.

**TTAHUB-412:** Date picker now allows you to move start date past the end date. If you do so, the delta between the two dates is preserved.

**TTAHUB-447:** The UI now handles export failures without bringing users to a white screen of death.

**TTAHUB-445:** Grantee search should only show grantees that have a grant that expired on or after 9/1/2020, or grantees that have an active grant.

This PR also includes an update of our version of node to stay compatible with cloud.gov buildpacks.

**TTAHUB-415:** Hide Total TTA Hrs controls and images from screen reader.

**TTAHUB-470:** Change "Program Type" column in Grants widget to "Program Type(s)" and show corresponding list of Grant Program Types.

## How to test

**TTAHUB-446:** Search for Grantees, Program Specialist should now show multiple. Viewing the Profile page of a Grantee should show a Grantee type and distinct list of Program specialists.

**TTAHUB-410:** Saving an AR or going to another page now shows a "fade-in" animation on the fixed save alert. Saving the AR on the Review Submit page shows an alert at the bottom of the page.

**TTAHUB-436:** Export CSV, confirm the correct message appears

**TTAHUB-412:** Create a new activity report, confirm that the date picker works correctly and that it preserves the delta between the dates

**TTAHUB-445:**: Verify that only grantees with active grants or grants that expired after 9/1/2020 appear in the search results

**TTAHUB-415:** Go to the dashboard page with a screen reader enabled. The only portion of the Total TTA hrs graph that should be read is the header, dates, and show table button.

**TTAHUB-470:** Search for any Grantee and view the Grants list on the Profile Page. The column "Program Type" should now be called "Program Type(s)" and give a distinct sorted list of Program Types (abbrev).

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-446
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-410
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-436
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-471
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-462
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-412
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-447
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-445
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-415
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-470

## Checklists

### Every PR


~- [ ] Boundary diagram updated~
~- [ ] Logical Data Model updated~
~- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions~

### Production Deploy


### After merge/deploy
